imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+from imap_processing.ialirt.l0.decom_ialirt import generate_xarray
+
+IALIRT_PACKET_LENGTH = 1464
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return imap_module_directory / "ialirt" / "packet_definitions" / "ialirt.xml"
+
+
+@pytest.fixture()
+def binary_packet_path(tmp_path):
+    """
+    Creates a binary file from the text packet data, which is more representative
+    of the actual operational environment. The binary file is deleted after the
+    test session.
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    binary_file_path = tmp_path / "file.ccsds"
+
+    with open(packet_path) as text_file, open(binary_file_path, "wb") as binary_file:
+        for line in text_file:
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1:]
+                # Convert hex to binary
+                binary_data = bytearray.fromhex("".join(hex_values))
+                binary_file.write(binary_data)
+                assert len(binary_data) * 8 == IALIRT_PACKET_LENGTH
+
+    yield binary_file_path
+
+    return binary_file_path
```

Comment:
Do we need both of these? I'm not sure which one would be needed to make sure the `tmp_path` doesn't get cleaned up too soon.